src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,127 +135,361 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+                        var titleForNesting = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0, data.Method, includeParameters: true);
+                        var titleCascading = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0_and_overrides_implementations, data.Method, includeParameters: true);
+                        codeAction = new CodeAction.CodeActionWithNestedActions(
+                            title: titleForNesting,
+                            ImmutableArray.Create(
+                                codeAction,
+                                new MyCodeAction(
+                                    title: titleCascading,
+                                    data.CreateChangedSolutionCascading)),
+                            isInlinable: true);
                     }
+
+                    builder.Add(codeAction);
+                }
+
+                return builder.ToImmutable();
+            }
+
+            ImmutableArray<CodeAction> NestByCascading()
+            {
+                var builder = ImmutableArray.CreateBuilder<CodeAction>(2);
+
+                var nonCascadingActions = ImmutableArray.CreateRange<CodeAction>(codeFixData.Select(data =>
+                {
+                    var title = GetCodeFixTitle(FeaturesResources.Add_to_0, data.Method, true);
+                    return new MyCodeAction(title: title, data.CreateChangedSolutionNonCascading);
+                }));
+
+                var cascading = codeFixData.Where(data => data.CreateChangedSolutionCascading != null);
+                var cascadingActions = ImmutableArray.CreateRange<CodeAction>(cascading.Select(data =>
+                {
+                    var title = GetCodeFixTitle(FeaturesResources.Add_to_0, data.Method, true);
+                    return new MyCodeAction(title: title, data.CreateChangedSolutionCascading);
+                }));
+
+                var aMethod = codeFixData.First().Method;
+                var nestedNonCascadingTitle = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0, aMethod, false);
+
+                builder.Add(new CodeAction.CodeActionWithNestedActions(nestedNonCascadingTitle, nonCascadingActions, isInlinable: false));
+
+                if (cascadingActions.Length > 0)
+                {
+                    var nestedCascadingTitle = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0_and_overrides_implementations, aMethod, false);
+                    builder.Add(new CodeAction.CodeActionWithNestedActions(nestedCascadingTitle, cascadingActions, isInlinable: false));
                 }
+
+                return builder.ToImmutable();
             }
+        }
+
+        private ImmutableArray<CodeFixData> PrepareCreationOfCodeActions(Document document, SeparatedSyntaxList<TArgumentSyntax> arguments, ImmutableArray<ArgumentInsertPositionData<TArgumentSyntax>> methodsAndArgumentsToAdd)
+        {
+            var builder = ImmutableArray.CreateBuilder<CodeFixData>(methodsAndArgumentsToAdd.Length);
```

Comment:
Preference is to use a type "ArrayBuilder" which can be pooled to help cut down on churn.